⬆️ draft support for laravel 6.*

lwiml opened the pull request:
Signed-off-by: Wim <wim@join.marketing>

Commit:
⬆️ draft support for laravel 6.*

Signed-off-by: Wim <wim@join.marketing>

roelofr commented on composer.json:
```diff
@@ -54,12 +55,12 @@
     "require": {
         "php": ">=7.1.0",
-        "laravel/framework": "5.5.*|5.6.*|5.7.*|5.8.*"
+        "laravel/framework": "5.5.*|5.6.*|5.7.*|5.8.*|6.*.*"
```
Since Laravel 6 is Semver, let's drop the `*`

```suggestion
        "laravel/framework": "5.5.*|5.6.*|5.7.*|5.8.*|^6.0"
```

roelofr commented on config/database-encryption.php:
```diff
@@ -3,7 +3,7 @@
  *
  * @author      Austin Heap <me@austinheap.com>
- * @version     v0.2.1
+ * @version     v0.2.2
```
This is *at least* a major unreleased change (v0.3.0), not a minor one.
```suggestion
 * @version     v0.3.0
```

roelofr commented on README.md:
```diff
@@ -63,7 +63,7 @@ directory. The most logical place to start are the [docs for the `HasEncryptedAt
 
-* Laravel: 5.5, 5.6, 5.7, or 5.8
+* Laravel: 5.5, 5.6, 5.7, 5.8, or 6.x
 * PHP: 7.1, 7.2, or 7.3
```
Since 7.1 is [already deprecated](https://www.php.net/supported-versions.php) and 7.2 is in security-only, maybe also look into PHP 7.4 support (add `allow-failure` CI tests for example)

lwiml replied:
Yes, That's good idea. For now I was just looking into my own case where i needed Laravel 6 and php 7.3 support.

roelofr replied:
I think that should be a separate PR. There will probably be some issues.

roelofr reviewed (commented):
Nothing really has changed. Are there any breaking changes so far?

Commit:
Improved support for Laravel 6.x

Commit:
Update composer.json

Co-Authored-By: Roelof <github@roelof.io>

lwiml commented:
> Nothing really has changed. Are there any breaking changes so far?

Thats right, everything seems to work fine so far, it's just the CI tests that i didn't get to succeed, because of the above dependency issues

Commit:
Update config/database-encryption.php

Co-Authored-By: Roelof <github@roelof.io>

Commit:
Added GitHub Actions

Commit:
Fixed tests not always supplying port number

Commit:
Cleaned up Travis CI file

- Don't double-install packages. Add new packages first and pull them
  all in at the same time.
- Removed uopz. It's not being used.
- Use `composer test`, which allows us to add additional tests there.

Commit:
Version 0.2.2 → 0.3.0

roelofr commented:
I had some suggestions, so I opened these in a [Pull Request on your fork](https://github.com/JoinMarketing/laravel-database-encryption/pull/1).

Commit:
Merge pull request #1 from gumbo-millennium/master

Improved Laravel 6 Support

leoneparise commented:
When will it be merged? I would like to use this lib on Laravel 6 without using my own repo.

roelofr commented on src/helpers.php:
```diff
@@ -96,3 +96,15 @@ function dbdecrypt(string $value): ?string
+     * @param int $value
+     * @return string|null
+     */
+    function str_random(int $value = 16): ?string
```
Why? Just replace it with `Illuminate\Support\Str::random()` if it's even being used anywhere.

roelofr commented:
> When will it be merged? I would like to use this lib on Laravel 6 without using my own repo.

Seconding this, I hope @austinheap has some time to look at this.

leoneparise commented:
> > When will it be merged? I would like to use this lib on Laravel 6 without using my own repo.
> 
> Seconding this, I hope @austinheap has some time to look at this.

Thanks @roelofr. Looking forward for next release.

dcnl1980 commented:
Any idea when it will be merged?

emielmolenaar commented:
A merge would be awesome :)

RobertCordes commented:
@emielmolenaar Yeah, especially since Laravel 7 is already live.

roelofr commented:
> 
> 
> @emielmolenaar Yeah, especially since Laravel 7 is already live.

Well, a major release every half year is something entirely different compared to the minor releases we had before.

potofcoffee commented:
Could you please merge this? This is the only thing keeping me back on Laravel 5.8

zeboe commented:
Any more news on this one already? In need of the 6 version as well.
Thanks!

eknip committed:
🔧 Adds support for future Laravel releases
